lib_test/test_httpaf.ml

```diff
@@ -154,9 +198,121 @@ module Server_connection = struct
       `Close (next_read_operation t);
   ;;
 
+  let test_synchronous_error () =
+    let request_string = "/ GET HTTP/1.1\r\n\r\n" in
+    let writer_woken_up = ref false in
+    let t = create ~error_handler synchronous_raise in
+    Alcotest.(check (of_pp Write_operation.pp_hum)) "Writer is in a yield state"
```

Comment:
It might be nice to lift and name the `of_pp ({Write,Read}_operation.pp_hum)` for brevity since it comes up a lot.

Reply:
Done.